apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/MetadataClient.cs

```diff
@@ -0,0 +1,303 @@
+        public virtual Task<Project> GetProjectMetadataAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Gets the value indicating whether the metadata server or an emulator for the server is available synchronously.
+        /// </summary>
+        /// <remarks>
+        /// Normally, the metadata server can only be accessed from an application running on a Google Compute Engine VM instance or
+        /// Google App Engine Flexible Environment, but if an metadata server emulator is available, this will also return true.
+        /// </remarks>
+        /// <returns>True if the normal metadata server or an emulator is available; false otherwise.</returns>
+        public virtual bool IsServerAvailable()
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Gets the value indicating whether the metadata server or an emulator for the server is available asynchronously.
+        /// </summary>
+        /// <remarks>
+        /// Normally, the metadata server can only be accessed from an application running on a Google Compute Engine VM instance or
+        /// Google App Engine Flexible Environment, but if an metadata server emulator is available, this will also return true.
+        /// </remarks>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <returns>A task containing true if the normal metadata server or an emulator is available; false otherwise.</returns>
+        public virtual Task<bool> IsServerAvailableAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            throw new NotImplementedException();
+        }
+
+        // TODO: Do we want separate events for when standard and custom metadata changes? Same question for project metadata.
+        // TODO: Should we suppress firing these when the custom metadata is updated from here?
+
+        /// <summary>
+        /// Occurs when the the instance's metadata has changed.
+        /// </summary>
+        /// <seealso cref="GetInstanceMetadata"/>
+        /// <seealso cref="GetInstanceMetadataAsync"/>
+        public virtual event EventHandler InstanceMetadataChanged
+        {
+            add { throw new NotImplementedException(); }
+            remove { throw new NotImplementedException(); }
+        }
+
+        /// <summary>
+        /// Occurs when the the instance's maintenance status has changed.
+        /// </summary>
+        /// <seealso cref="GetMaintenanceStatus"/>
+        /// <seealso cref="GetMaintenanceStatusAsync(CancellationToken)"/>
+        /// <seealso cref="SchedulingMetadata.MaintenanceEventBehavior"/>
+        public virtual event EventHandler<MaintenanceStatus> MaintenanceStatusChanged
```

Comment:
Good point. Looking back, it seems we even discussed letting the caller setup a listener for a particular path/value, but that must have fallen through the cracks. So I guess this key in your proposed method signature would be the relative path (everything after http://metadata.google.internal/computeMetadata/v1/)? So one example might be "instance/network-interfaces/1/forwarded-ips/" - is this correct? And the string result would be the changed value (or current value if the timeout elapses)?